File: dev/app_ios/StayOn/Models/SleepPlan.swift
import Foundation
import SwiftData

@Model
final class SleepPlan {
    @Attribute(.unique) var id: UUID
    var bedtime: Date
    var preReminderMinutes: Int
    var snoozeIntervalMinutes: Int
    var maxSnoozeCount: Int
    var isActive: Bool
    var updatedAt: Date

    init(
        id: UUID = UUID(),
        bedtime: Date,
        preReminderMinutes: Int = 30,
        snoozeIntervalMinutes: Int = 10,
        maxSnoozeCount: Int = 1,
        isActive: Bool = true,
        updatedAt: Date = .now
    ) {
        self.id = id
        self.bedtime = bedtime
        self.preReminderMinutes = preReminderMinutes
        self.snoozeIntervalMinutes = snoozeIntervalMinutes
        self.maxSnoozeCount = maxSnoozeCount
        self.isActive = isActive
        self.updatedAt = updatedAt
    }
}

File: dev/app_android/app/src/main/java/com/stayon/app/domain/model/SleepPlan.kt
package com.stayon.app.domain.model

data class SleepPlan(
    val id: String = java.util.UUID.randomUUID().toString(),
    val bedtimeHour: Int = 23,
    val bedtimeMinute: Int = 30,
    val preReminderMinutes: Int = 30,
    val snoozeIntervalMinutes: Int = 10,
    val maxSnoozeCount: Int = 1,
    val isActive: Boolean = true,
    val updatedAt: Long = System.currentTimeMillis()
)
